api/mysagw/case/serializers.py

```diff
@@ -72,3 +73,36 @@ def create(self, validated_data):
         )
 
         return instance
+
+
+class CaseTransferSerializer(drf_serializers.Serializer):
+    removable_assignees = drf_serializers.ListField(
+        child=drf_serializers.PrimaryKeyRelatedField(queryset=Identity.objects.all())
+    )
+    new_assignees = drf_serializers.ListField(
+        child=drf_serializers.PrimaryKeyRelatedField(queryset=Identity.objects.all()),
+        allow_empty=False,
+    )
+    case_ids = drf_serializers.ListField(
+        child=drf_serializers.UUIDField(), allow_empty=False
+    )
+
+    def create(self, validated_data):
+        new_accesses = []
+        for case_id in validated_data["case_ids"]:
+            for new_assignee in validated_data["new_assignees"]:
+                if not models.CaseAccess.objects.filter(
+                    identity=new_assignee, case_id=case_id
+                ).exists():
+                    new_accesses.append(  # noqa: PERF401
+                        models.CaseAccess.objects.create(
+                            case_id=case_id, identity=new_assignee
+                        )
+                    )
```

Comment:
Could probably be simplified by calling `CaseAccess.objects.update_or_create()` instead

Reply:
Done :+1: I went with `get_or_create()`, as there is nothing to update.